### PR TITLE
Blog - Konfig - How Earthly Solved Our CI Problem

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@adamgordonbell [Preview link](https://65a79dfed410ec913e67f08f--ephemeral-narwhal-3ac3b2.netlify.app/blog/konfig/)

[Airtable link](https://airtable.com/appbtxxY2KaPio7dV/tblsY8AOw3qPBIUdZ/viwBflqOY7OlQqiNg/recacolsXBbz39OAS?blocks=hide)

File: blog/_posts/2024-01-22-konfig.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+internal-links:
+ - ci problem solved
+ - how earthly helps with ci problems
+ - solving ci problems with earthly
+ - earthly the problem solver
+excerpt: |
+    In this guest post, the Konfig team discusses how they solved their complex Continuous Integration challenges with Earthly, sharing insights valuable for any software development team.
+---
+
```
[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2024-01-22-konfig.md:16]: should I remove this from excerpt?

File: blog/_posts/2024-01-22-konfig.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+![Img]({{site.images}}{{page.slug}}/img3.png)\
+
+The value of a continuous integration pipeline in software development is clear. It allows engineers to test every code change, ensuring deployments are free from regressions in functionality and performance. At Konfig, a startup creating SDKs in many coding languages from our clients' OpenAPI specs, not having Continuous Integration was a major issue. Code generation software tends to be especially fickle, and many of our code paths are shared across generators for different coding languages, making it easy to introduce bugs. When we made revisions for one customer, it wasn't uncommon for these changes to be incompatible for another customer. We badly needed CI -- and quickly, too. Thus we embarked on (what we thought would be) the treacherous journey of building a CI pipeline that could accommodate our architecture.
+
+![Img]({{site.images}}{{page.slug}}/img1.png)\
```
Comment (adamgordonbell) [inline blog/_posts/2024-01-22-konfig.md:21]: I'm not sure about this image here, but we could try having the intro underneath it italics and maybe that would work? @thtmnisamnstr  thoughts.

So underneath it:

_In this guest post, the Konfig team discusses how they solved their complex Continuous Integration challenges with Earthly, sharing insights valuable for any software development team._

We probably need a link out to Konfig somewhere as well.

File: blog/_posts/2024-01-22-konfig.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamgordonbell) [reply blog/_posts/2024-01-22-konfig.md:16]: I mean, it does work as the excerpt, so I see why you put it there. I think that works. We just also need it in the article.

[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2024-01-22-konfig.md:16]: ok

**Comment by msali123** [comment]
@adamgordonbell [Updated link](https://65a7ee0a4dbeb60e122ef6d9--ephemeral-narwhal-3ac3b2.netlify.app/blog/konfig/)

File: blog/_posts/2024-01-22-konfig.md (hunk 2 above)
Comment (thtmnisamnstr) [reply blog/_posts/2024-01-22-konfig.md:23]: I agree on having the intro under the image in italics. The first mention of Konfig in the post, in the intro, should be linked.

File: blog/_posts/2024-01-22-konfig.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## The Problem: Multiple Languages and Components
+
+From the get-go, we presumed that building an efficient CI pipeline for Konfig would not be straightforward. None of us had any experience creating a CI pipeline for a multi-service architecture. Our product naturally interacts with many languages, necessitating the deployment of different services tailored to each language. For instance, certain functionalities, like formatting, require environments that support their specific language. . Despite our complicated architecture, we wanted our CI to be speedy and cost-effective. As a startup, we were moving fast, and could not afford for build times to stretch into hours. This was a tricky problem.
```
Comment (thtmnisamnstr) [inline blog/_posts/2024-01-22-konfig.md:27]: Can you fix the double periods in this paragraph, "specific language. . Despite"?

**Comment by msali123** [comment]
@adamgordonbell [Updated Link](https://65a82c6d817b280c09c5262b--ephemeral-narwhal-3ac3b2.netlify.app/blog/konfig/)

**Review by thtmnisamnstr — APPROVED** [review]

**Comment by eddiechayes** [comment]
Awesome, looks great! Thank you guys!

**Pull request merged**